panic during syncusers

Encountered this panic:

```
# wwctl container syncuser rocky-8-split
panic: runtime error: index out of range [2] with length 1

goroutine 1 [running]:
github.com/hpcng/warewulf/internal/pkg/container.createPasswdMap({0xd48675, 0xb})
	/builddir/build/BUILD/warewulf-4.3.0/internal/pkg/container/syncuids.go:225 +0x632
github.com/hpcng/warewulf/internal/pkg/container.SyncUids({0x7ffff05f8183, 0xd}, 0x1)
	/builddir/build/BUILD/warewulf-4.3.0/internal/pkg/container/syncuids.go:43 +0x7b
github.com/hpcng/warewulf/internal/app/wwctl/container/syncuser.CobraRunE(0x15e64e0, {0xc00006e520, 0x1, 0x1})
	/builddir/build/BUILD/warewulf-4.3.0/internal/app/wwctl/container/syncuser/main.go:17 +0x6e
github.com/spf13/cobra.(*Command).execute(0x15e64e0, {0xc00006e4f0, 0x1, 0x1})
	/builddir/build/BUILD/warewulf-4.3.0/vendor/github.com/spf13/cobra/command.go:850 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x15e3d80)
	/builddir/build/BUILD/warewulf-4.3.0/vendor/github.com/spf13/cobra/command.go:958 +0x3ad
github.com/spf13/cobra.(*Command).Execute(...)
	/builddir/build/BUILD/warewulf-4.3.0/vendor/github.com/spf13/cobra/command.go:895
main.main()
	/builddir/build/BUILD/warewulf-4.3.0/cmd/wwctl/main.go:13 +0x25
```

Using the RPM from GitHub.

```
# wwctl version
4.3.0-1.git_235c23c
```

I'm still investigating the circumstances.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -116,6 +116,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Don't show an error if image files for containers can't be found. #933
 - Make configured paths available in overlays as `.Path` #960
 - Support importing containers with symlinked `/bin/sh` #797
+- Don't panic on malformed passwd #527
 
 ## [4.4.0] 2023-01-18
 

--- a/internal/pkg/container/syncuids.go
+++ b/internal/pkg/container/syncuids.go
@@ -167,7 +167,10 @@ func (db syncDB) read(fileName string, fromContainer bool) error {
 		for fileScanner.Scan() {
 			line := fileScanner.Text()
 			fields := strings.Split(line, ":")
-
+			if len(fields) != 7 {
+				wwlog.Debug("malformed line in passwd: %s", line)
+				continue
+			}
 			name := fields[0]
 			if name == "" {
 				continue

--- a/internal/pkg/container/syncuids_test.go
+++ b/internal/pkg/container/syncuids_test.go
@@ -258,3 +258,13 @@ func Test_differ(t *testing.T) {
 	assert.False(t, entry.match())
 	assert.True(t, entry.differ())
 }
+
+func Test_malformed_passwd(t *testing.T) {
+	hostInput := `"testuser1:x:1001:1001::/home/testuser:/bin/bash"
+	asdf`
+	hostFileName := writeTempFile(t, hostInput)
+	defer os.Remove(hostFileName)
+	db := make(syncDB)
+	err := db.readFromHost(hostFileName)
+	assert.NoError(t, err)
+}

